python/ray/exceptions.py

```diff
@@ -309,27 +348,34 @@ def __init__(self, cause: Union[RayTaskError, ActorDiedErrorContext] = None):
                     "The actor never ran - it was cancelled before it started running."
                 )
             if cause.preempted:
-                self._preempted = True
+                preempted = True
                 error_msg_lines.append(
                     "\tThe actor's node was killed by a spot preemption."
                 )
-            self.error_msg = "\n".join(error_msg_lines)
-            self.actor_id = ActorID(cause.actor_id).hex()
+            error_msg = "\n".join(error_msg_lines)
+            actor_id = ActorID(cause.actor_id).hex()
+        super().__init__(actor_id, error_msg, actor_init_failed, preempted)
 
-    @property
-    def actor_init_failed(self) -> bool:
-        return self._actor_init_failed
+    @staticmethod
+    def from_task_error(task_error: RayTaskError):
+        return ActorDiedError(task_error)
 
-    def __str__(self) -> str:
-        return self.error_msg
 
-    @property
-    def preempted(self) -> bool:
-        return self._preempted
+@DeveloperAPI
+class ActorUnavailableError(RayActorError):
+    """Raised when the actor is temporarily unavailable but may be available later."""
+
+    def __init__(self, error_message: str, actor_id: Optional[bytes]):
+        actor_id = ActorID(actor_id).hex() if actor_id is not None else None
+        error_msg = (
+            f"The task couldn't be completed because the actor {actor_id} is "
+            f"unavailable: {error_message}.\n\nThis task is automatically "
+            "retried if `max_retries` is set and there are retries remaining."
```

Comment:
Again, seems strange that we would raise this exception if there are retries enabled? The point of retries is to not have to handle the exceptions at the user level.

Also, we cannot guarantee that "the task couldn't be completed", right? It may have executed and then the connection closed?

Reply:
This exception is raised when the task has ran out of task_retries. Removed the last line as misleading.

"the task couldn't be completed" is indeed not the best word, since the task may or may not be completed. Removed to focus more on